### DISCUSSION TITLE
@entangle not binding both Alpine and Livewire when using model property

### DISCUSSION THREAD

**Discussion by computerwizard666** [livewire/livewire#5326] [7. Bugs (archived)]
Hi there! I'm having a bit of an issue with entangle on a model...

I'm using:

"alpinejs": "^3.10.4"
"livewire/livewire": "^2.10"

This is my component (Page is a model with one fillable attribute - 'title'):

```php
class Editor extends Component
{
    public Page $page;

    protected $rules = [
        'page.title' => 'required',
    ];

    public function mount()
    {
        $this->page = Page::first();
    }
}
```

and this is my Livewire blade:

```html
<div x-data="{ page: @entangle('page') }">
    <input type="text" wire:model="page.title" />

    <input type="text" x-model="page.title" />
</div>
```

changing the first input changes the second one too (as expected) but changing the second input results in:

```
Cannot bind property [page] without a validation rule present in the [$rules] array on Livewire component: [editor].
```

I added an empty rule `'page' => ''` in the component just to try it out and then got this:

```
Cannot assign array to property App\Http\Livewire\Editor::$page of type App\Models\Page
```

I have tried the same setup but with just a string instead of a Model and it works fine so the issue happens only when "x-model" is used on a property of an entangled model

**Comment by ramdoni** [comment]
Hi @computerwizard666  try this 

`public $page = ["title"=>"","description"=>""];`

`public function mount()
    {
        $this->page = Page::first()->toArray();
    }`

**Comment by michael-rubel** [comment]
Avoid using models as component properties. Use model attributes (as primitive types) instead.
You can map fields dynamically from the model using `$this->fill()` or [Loop Functions](https://github.com/michael-rubel/laravel-loop-functions) package.

**Comment by joshhanley** [comment]
@computerwizard666 `@entangle` doesn't support deep watching. As such you should entangle the individual properties of `page` instead of `page` itself.

Like
```blade
<div x-data="{ title: @entangle('page.title') }">
    <input type="text" x-model="title" />
</div>
```

Hope this helps!
